lib/assets/javascripts/cartodb/table/menu_modules/filters/histogram.js

```diff
@@ -255,8 +256,18 @@
           var upperLimit = this.model.get('upper_limit');
           if (value < lowerLimit) { value = lowerLimit; }
           if (value > upperLimit) { value = upperLimit; }
-          if (isMin && value > upper) { value = upper; }
-          if (!isMin && value < lower) { value = lower; }
+          if (isMin && value > upper) {
+            temp = upper;
+            upper = value;
+            value = temp;
+            this.model.set('upper', upper);
```

Comment:
Yeah I like this better -- its more concise, saves a temp var and is at least as clear. Changed and added comments.